src/palindrom.js

```diff
@@ -248,6 +248,52 @@ var Palindrom = (function() {
       }
     });
   }
+  /**
+   * Traverses/checks value looking for out-of-range numbers, throws a RangeError if it finds any
+   * @param {*} val value 
+   * @param {string} direction the inspected patch direction "outgoing"|"incoming"
+   */
+  PalindromNetworkChannel.prototype.findRangeErrors = function(val, direction) {
+    var type = typeof val;
+    if (type == 'object') {
+      for (var prop in val) {
+        if (val.hasOwnProperty(prop)) {
+          this.findRangeErrors(val[prop], direction);
+        }
+      }
+    } else if (
+      type === 'number' &&
+      (val > 9007199254740991 || val < -9007199254740991)
+    ) {
+      if (direction === 'outgoing') {
+        this.palindrom.onOutgoingPatchValidationError(
+          new RangeError(
+            `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in an ${direction} patch, value is ${val}`)
+        );
+      } else {
+        this.palindrom.onIncomingPatchValidationError(
+          new RangeError(
+            `A number that is either bigger than Number.MAX_INTEGER_VALUE or smaller than Number.MIN_INTEGER_VALUE has been encountered in an ${direction} patch, value is: ${val}`)
+        );
```

Comment:
Instead of sending direction as `string`, pass the reference to `onOutgoingPatchValidationError` or `onIncomingPatchValidationError` error callback.